### ISSUE TITLE
🐛Bug: encountering `ERR_SSL_PROTOCOL_ERROR` on melt-ui.com regularly

### ISSUE THREAD

**Issue by mcmxcdev** [melt-ui/melt-ui#746]
**Describe the bug**
I am using melt-ui for multiple months now and most of the time, I am unable to use the website at all.

There are times when the website works again without any action on my side, but I don't see any pattern.

It's very unclear what this issue is related to, but other users reported it on Discord as well, so it seems to be melt-ui specific.

**To Reproduce**
1. Go to https://www.melt-ui.com
2. See the following error:

```
This site can’t provide a secure connection.
www.melt-ui.com sent an invalid response.
ERR_SSL_PROTOCOL_ERROR
```

**Expected behavior**
https://www.melt-ui.com should work like any other website.

**Screenshots**
Firefox browser probably shows the most detailed error message about it:

![image](https://github.com/melt-ui/melt-ui/assets/16797721/628efded-6a15-417e-b502-5bce039c1d0a)

**Desktop (please complete the following information):**
 - OS: Ubuntu 23.10
 - Browser: Brave, Firefox, etc.

**Smartphone (please complete the following information):**
 - Device: iPhone 11
 - OS: 17.0.3
 - Browser Brave, Safari, etc.

**Comment by alexanderl19** [comment]
I've been running into this issue too for the past few days.

Here's a mirror of the site for the time being: https://melt-ui.alexanderliu.dev/

**Comment by mcmxcdev** [comment]
Thanks so much for providing a link, it works for me and I will use it until this issue is resolved :wink:

**Comment by TGlide** [comment]
What regions are you in, if I may ask @mcmxcdev @alexanderl19 ?

**Comment by mcmxcdev** [comment]
I am in America/Denver timezone.

**Comment by alexanderl19** [comment]
America/Los_Angeles

**Comment by TGlide** [comment]
Thank you. I've updated a small configuration, could you check if it still happens? I'm not able to find the SSL error even when using a VPN

**Comment by alexanderl19** [comment]
Still getting the same SSL error, even in an incognito window.

Best I can tell, the home pages loads fine, but after visiting one of the component pages, the entire site errors.
![Firefox 2023-11-23 at 04 10 32@2x](https://github.com/melt-ui/melt-ui/assets/41758627/fe302ce9-b68a-4cfe-b62f-cb32e541e34f)

Interestingly, I've been using the deploy I have of main ([meltui.alexanderliu.dev](https://meltui.alexanderliu.dev/)), and it's been working without issue. It's deployed on Vercel.

**Comment by alexanderl19** [comment]
I just visited the site with a VPN (ProtonVPN) and it loads as expected even after navigating around.

Also, I originally deployed the develop branch to Vercel, and experienced the same SSL issues.

**Comment by leerobert** [comment]
I'm also getting the error on firefox,chrome, mac os

An error occurred during a connection to www.melt-ui.com. SSL received a record that exceeded the maximum permissible length.

Error code: SSL_ERROR_RX_RECORD_TOO_LONG

**Comment by TGlide** [comment]
> I'm also getting the error on firefox,chrome, mac os
> 
> An error occurred during a connection to [www.melt-ui.com](http://www.melt-ui.com). SSL received a record that exceeded the maximum permissible length.
> 
> Error code: SSL_ERROR_RX_RECORD_TOO_LONG

Thanks for sharing. Does the same happen with `https://stg.melt-ui.com`?

**Comment by TGlide** [comment]
This has also been super hard to debug 😓 I can't for the life of me reproduce the SSL error, and tests show everything's fine (https://www.ssllabs.com/ssltest/analyze.html?d=melt-ui.com), when it's obviously not. If anyone has any clues how to solve it, I'd appreciate it.

**Comment by leerobert** [comment]
Yeah it intermittently works for me. For example, right now I can't connect to https://melt-ui.com but https://stg.melt-ui.com does work.

**Comment by GhostWalker562** [comment]
Running into this issue as well.

**Comment by alexanderl19** [comment]
> > I'm also getting the error on firefox,chrome, mac os
> > An error occurred during a connection to [www.melt-ui.com](http://www.melt-ui.com). SSL received a record that exceeded the maximum permissible length.
> > Error code: SSL_ERROR_RX_RECORD_TOO_LONG
> 
> Thanks for sharing. Does the same happen with `https://stg.melt-ui.com`?

Yes

**Comment by TGlide** [comment]
> Yes, but ERR_SSL_PROTOCOL_ERROR on Chrome
> 
> Also, my deploy of main ([meltui.alexanderliu.dev](https://meltui.alexanderliu.dev/)) stopped working.

Weird, it works for me... are you hosting it on vercel by any chance?

**Comment by joekrump** [comment]
Hitting this as well in Firefox, Chrome and Safari on MacOS 14.1.2

From what firefox is telling me, the root cause is `SSL_ERROR_RX_RECORD_TOO_LONG`

<img width="941" alt="image" src="https://github.com/melt-ui/melt-ui/assets/3317231/a4135a38-6ad3-4f52-b9f9-8e6ce62e74f0">

**Comment by joekrump** [comment]
> This has also been super hard to debug 😓 I can't for the life of me reproduce the SSL error, and tests show everything's fine (https://www.ssllabs.com/ssltest/analyze.html?d=melt-ui.com), when it's obviously not. If anyone has any clues how to solve it, I'd appreciate it.

Looking at what this is testing against, I see that not many tests are run with MacOS. I'm wondering if everyone experiencing the issue is running Mac and possibly a m1 or m2 processor. I'm in the camp of Mac + m2 at the moment.

**Comment by TGlide** [comment]
> Have you tested yourself using a Mac + the latest Chrome for example?

Yup, I have a Mac M2

**Comment by alexanderl19** [comment]
M1

**Comment by mcmxcdev** [comment]
I am using Ubuntu 23.10, so probably not OS-related.

**Comment by mcmxcdev** [comment]
I found a temporary solution that works for me: connecting to any server on Proton VPN, regardless of the country.

It's tedious though and when disconnecting, the website is down again.

**Comment by Avevrr** [comment]
On my end I was able to determine that the site is being blocked by "Xfinity xFi Advanced Security" for the reason of "Phishing/Other Frauds". I reported the issue to [spa.xfinity.com](https://spa.xfinity.com). I asked them why the site is being blocked and if the site may be on another provider's  "deny list" that Comcast is using.

![Screenshot_20231208-211010-381](https://github.com/melt-ui/melt-ui/assets/47205349/273e0989-5210-4aaa-8519-9203eb1f94b7)

**Comment by huntabyte** [comment]
Thanks for making us aware of this @Avevrr!

The only thing I could see as it marking it as phishing is: 
![CleanShot 2023-12-08 at 23 45 10@2x](https://github.com/melt-ui/melt-ui/assets/64506580/98de9496-c8bf-40d6-b9a1-8b2f49b27e5a)

Unless the previous owner of the domain (if any) participated in questionable behavior. Either way I'm interested to see what they say.

**Comment by jesperordrup** [comment]
Is this a melt-ui related issue? I mean is this an issue that ppl think is with the use of melt-ui and therefore relevant in the repo?

Or is it "just" a website SSL problem ?

**Comment by TGlide** [comment]
Hey all. I've decided to give Cloudflare a go. Can you all tell me if this link is working for you? https://cf.melt-ui.com

**Comment by alexanderl19** [comment]
> Thanks for making us aware of this @Avevrr!
> 
> The only thing I could see as it marking it as phishing is: ![CleanShot 2023-12-08 at 23 45 10@2x](https://private-user-images.githubusercontent.com/64506580/289234272-98de9496-c8bf-40d6-b9a1-8b2f49b27e5a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDIxNTQ5MDQsIm5iZiI6MTcwMjE1NDYwNCwicGF0aCI6Ii82NDUwNjU4MC8yODkyMzQyNzItOThkZTk0OTYtYzhiZi00MGQ2LWI5YTEtOGIyZjQ5YjI3ZTVhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjA5VDIwNDMyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxMmFlNTQ5MmIzODAzODlmM2Y3YWU3ZmQ5YzViNjcwOTNjYzMyOTllZGY1MzRmNjJkZjAyNjM3OTRmZDU0YzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5JGH2M8FSZ5wC3jTYslfA5Y1pO9bRvyA5kLY9KIW37A)
> 
> Unless the previous owner of the domain (if any) participated in questionable behavior. Either way I'm interested to see what they say.

I wonder if it is just an ISP issue, since both the sites load when using ProtonVPN or on my mobile hotspot.

**Comment by Avevrr** [comment]
On my end, with "Xfinity xFi Advanced Security" enabled

- cf.melt-ui.com loads
- melt-ui.com and www.melt-ui.com do not load.  Firefox reports Error code: SSL_ERROR_RX_RECORD_TOO_LONG but I see new entries in the Xfinity Advanced Security admin as "blocked and is classified as: Phishing/Other Frauds".

**Comment by TGlide** [comment]
I wonder, from people that have the SSL error on the original website, is anyone *not* using Xfinity?

**Comment by joekrump** [comment]
> I wonder, from people that have the SSL error on the original website, is anyone _not_ using Xfinity?

I'm using a canadian provider call [Rogers](https://www.rogers.com/)

**Comment by TGlide** [comment]
> I'm using a canadian provider call [Rogers](https://www.rogers.com/)

And is the original website failing for you? What about the Cloudflare link?

If you find out whether it's marked as a phishing website, please let us know, it'd be a huge help.

**Comment by alexanderl19** [comment]
> I wonder, from people that have the SSL error on the original website, is anyone _not_ using Xfinity?

[Cox](https://www.cox.com/residential/home.html)

**Comment by TGlide** [comment]
# UPDATE: Melt UI is now running on Cloudflare

The main domain, `https://melt-ui.com`, is now officially being hosted on Cloudflare. Can everyone confirm if it's working for them?

**Comment by leerobert** [comment]
Can confirm I can load it without issues.

**Comment by mcmxcdev** [comment]
Unfortunately, for me, it still only works through VPN.

**Comment by mcmxcdev** [comment]
https://cf.melt-ui.com/ doesn't work for me either.

I am using Xfinity, unfortunately can't verify right now if "Xfinity xFi Advanced Security" is enabled.

**Comment by TGlide** [comment]
> https://cf.melt-ui.com/ doesn't work for me either.
> 
> I am using Xfinity, unfortunately can't verify right now if "Xfinity xFi Advanced Security" is enabled.

I see, Xfinity likely blacklisted the website. Last question, does https://melt-ui.pages.dev/ work?

**Comment by huntabyte** [comment]
It has to be an issue with the domain name itself being on some blocklists.

**Comment by TGlide** [comment]
Probably. I'd recommend changing DNS provider to something less obnoxious, such as https://developers.cloudflare.com/1.1.1.1/

**Comment by TGlide** [comment]
In the meanwhile, we can **try** and detect what ISPs are blocking us, try and email them.

**Comment by Avevrr** [comment]
I did notice that Search does not work on the site I just loaded from melt-ui.com. It will show "No results" regardless of what is typed.

I am pretty sure its just the cf hosted site. I had noticed it at some point over the weekend (just on the cf subdomainwhile the main site search worked).

 See screenshot regarding pagefind.js.

I can create a new issue if that's better

![Screenshot from 2023-12-12 18-00-16](https://github.com/melt-ui/melt-ui/assets/47205349/bb4604be-7b4e-414f-b6cc-9aee6257ec80)

**Comment by TGlide** [comment]
Yeah, I noticed that too. I don't think a new issue is needed, I'll fix it in a jiffy :)

**Comment by mcmxcdev** [comment]
> * [melt-ui.pages.dev](https://melt-ui.pages.dev/) -> works
> * [cf.melt-ui.com](https://cf.melt-ui.com/) -> didnt work in the morning when I checked first, but now does for some reason
> * [melt-ui.com](https://melt-ui.com/) -> doesnt work

Not sure what changed but I can access all 3 of them now just fine! 

I had a chance to look into Xfinity logs too and can confirm that the website was flagged as fraud, same as https://github.com/melt-ui/melt-ui/issues/746#issuecomment-1848097452

**Comment by mustafa0x** [comment]
www causes a redirect loop.

```
$> curl -I https://www.melt-ui.com/docs/introduction
HTTP/2 308
location: https://www.melt-ui.com/docs/introduction
```

**Comment by TGlide** [comment]
> www causes a redirect loop.
> 
> ```
> $> curl -I https://www.melt-ui.com/docs/introduction
> HTTP/2 308
> location: https://www.melt-ui.com/docs/introduction
> ```

Fixed! Hard refresh and you should no longer see the issue

**Comment by joekrump** [comment]
> # UPDATE: Melt UI is now running on Cloudflare
> The main domain, `https://melt-ui.com`, is now officially being hosted on Cloudflare. Can everyone confirm if it's working for them?

Seems to have fixed things for me! Thanks @TGlide

**Comment by mcmxcdev** [comment]
Looks like this was fixed for good, thanks to @TGlide!

I will close this but if anyone is still experience this, we can reopen it.
